src/test/lib/utils/meta_tables/meta_temporary_memory_usage_table_test.cpp

```diff
@@ -0,0 +1,109 @@
+#include "base_test.hpp"
+
+#include "hyrise.hpp"
+#include "storage/table.hpp"
+#include "utils/meta_tables/meta_temporary_memory_usage_table.hpp"
+
+namespace opossum {
+
+class MetaTemporaryMemoryUsageTest : public BaseTest {
+ protected:
+  std::shared_ptr<AbstractMetaTable> meta_temporary_memory_usage_table;
+  void SetUp() override {
+    meta_temporary_memory_usage_table = std::make_shared<MetaTemporaryMemoryUsageTable>();
+  }
+  void TearDown() override {
+    Hyrise::reset();
+  }
+
+  const std::shared_ptr<Table> generate_meta_table(const std::shared_ptr<AbstractMetaTable>& table) const {
+    return table->_generate();
+  }
+};
+
+TEST_F(MetaTemporaryMemoryUsageTest, IsImmutable) {
+  EXPECT_FALSE(meta_temporary_memory_usage_table->can_insert());
+  EXPECT_FALSE(meta_temporary_memory_usage_table->can_update());
+  EXPECT_FALSE(meta_temporary_memory_usage_table->can_delete());
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, TableName) {
+  EXPECT_EQ(meta_temporary_memory_usage_table->name(), "temporary_memory_usage");
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, EmptyTableGeneration) {
+  const auto expected_table =
+      Table::create_dummy_table(TableColumnDefinitions{{"operator_type", DataType::String, false},
+                                                       {"operator_data_structure", DataType::String, false},
+                                                       {"timestamp", DataType::Long, false},
+                                                       {"amount", DataType::Long, false}});
+  EXPECT_TABLE_EQ_UNORDERED(generate_meta_table(meta_temporary_memory_usage_table), expected_table);
+}
+
+TEST_F(MetaTemporaryMemoryUsageTest, TableGeneration) {
+  // record some allocations and deallocations
+  auto& memory_resource_manager = Hyrise::get().memory_resource_manager;
+  memory_resource_manager.enable_temporary_memory_tracking();
+  auto memory_resource_1 = memory_resource_manager.get_memory_resource(OperatorType::Mock, "my_data_structure_1");
+  auto memory_resource_2 = memory_resource_manager.get_memory_resource(OperatorType::Mock, "my_data_structure_2");
+  auto mem1 = memory_resource_1->allocate(10);
+  auto mem2 = memory_resource_1->allocate(20);
+  auto mem3 = memory_resource_2->allocate(5);
+  memory_resource_1->deallocate(mem1, 10);
+  memory_resource_1->deallocate(mem2, 20);
+  memory_resource_2->deallocate(mem3, 5);
+
+  const auto meta_table = generate_meta_table(meta_temporary_memory_usage_table);
+
+  ASSERT_EQ(meta_table->row_count(), 6);
+
+  const auto timestamp_ns = std::chrono::nanoseconds{std::chrono::system_clock::now().time_since_epoch()}.count();
+
+  // Check individual rows (we can't simply compare to a pre-built table
+  // because the timestamp values will vary between test runs).
```

Comment:
If you want to avoid the problem mentioned, you could introduce a Hyrise wise clock singleton that you can control. For release builds, it would return the usual time from the system clock and for tests it would return a monotonically increasing counter to return different but deterministic results on each call.

Reply:
Thanks for the suggestion. We would like to keep the behaviour of the code during testing time as close to the production behaviour as possible. Thus, we would opt for the more complex test code in this case.